test/server_helper_test.js

```diff
@@ -29,83 +31,151 @@ contract('ServerHelper', function (accounts) {
+                { relayUrl:"url5", MinGasPrice: 1, Ready:true, stake:1, unstakeDelay:1 },
+                { relayUrl:"url6", Ready:false, stake:1, unstakeDelay:1 },
+                { relayUrl:"url7", MinGasPrice: 1, Ready:true, stake:1, unstakeDelay:1 },
+            ]
 
 
-        mockRelays.forEach(r => r.relayUrl = JSON.stringify(r))
+            mockRelays.forEach(r => r.relayUrl = JSON.stringify(r))
 
-        let mockHttpWrapper = new MockHttpWrapper( mockRelays )
+            let mockHttpWrapper = new MockHttpWrapper( mockRelays )
 
-        let pinger = new serverHelper.ActiveRelayPinger(mockRelays, mockHttpWrapper, 100)
+            let pinger = new serverHelper.ActiveRelayPinger(mockRelays, mockHttpWrapper, 100)
 
-        //should skip the bad relays, 3 at a time, and reach relay 5
-        let r = await pinger.nextRelay()
-        //validate its "url5" that got returned (the other were rejected)
-        assert.equal("url5", JSON.parse(r.relayUrl).relayUrl )
-        //make sure we totally tried exactly 6 relays (we ping in triplets)
-        assert.equal(6, mockHttpWrapper.pinged )
+            //should skip the bad relays, 3 at a time, and reach relay 5
+            let r = await pinger.nextRelay()
+            //validate its "url5" that got returned (the other were rejected)
+            assert.equal("url5", JSON.parse(r.relayUrl).relayUrl )
+            //make sure we totally tried exactly 6 relays (we ping in triplets)
+            assert.equal(6, mockHttpWrapper.pinged )
 
-    })
+        })
+    });
 
+    describe('with mock relay hub', function () {
+        // let minStake = 1.5e17
+        // let minDelay = 10
+    
+        const mockRelayAddedEvents = [
+            { relay: '1' },
+            { relay: '2' },
+            { relay: '3' },
+            { relay: '4', unstakeDelay: 5 },
+            { relay: '5', stake: 1e17, transactionFee: 1e5 },
+            { relay: '6', stake: 3e17, transactionFee: 1e9 },
+            { relay: '7', transactionFee: 1e7 },
+        ].map(relay => ({ 
+            event: 'RelayAdded', 
+            returnValues: Object.assign({}, { 
+                transactionFee: 1e10, 
+                relayUrl: `url-${relay.relay}`, 
+                stake: 2e17, 
+                unstakeDelay: 100
+            }, relay)
+        }));
+
+        beforeEach('set mock relay hub', function () {
+            this.originalRelayHub = serverHelper.relayHubInstance;
+            this.mockRelayHub = { getPastEvents: () => mockRelayAddedEvents };
+            serverHelper.setHub(this.mockRelayHub);
+        });
+
+        afterEach('restore original relay hub', function () {
+            serverHelper.setHub(this.originalRelayHub);
+        });
+
+        it("should use default strategy for filtering and sorting relays", async function() {
+            const relays = await serverHelper.fetchRelaysAdded();
+            assert.deepEqual(relays.map(r => r.address), ['7', '6', '1', '2', '3']);
```

Comment:
it would help to add the rationale why that was the result
(I know, our tests are not well documented either... Just took me some time to understand WHY this is the correct result..)